### PR TITLE
generalize Renesas SCE Protected port

### PR DESCRIPTION
**Pull request by miyazakh**
# Description

Generalize SCE Protect port code as much as possible. 
SCE Protected FSP module now is treated like a one of Secure module of FSP.
Update key data

# Testing

Run tests including crypt, benchmark and client on RA6M4 with SCE Protected 

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by miyazakh** [comment]
retest this please

File: IDE/Renesas/e2studio/RA6M4/test/src/test_main.c [hunk 1]
```diff
@@ -381,22 +396,26 @@ void sce_test(void)
...
-
-          memset(info[i].name, 0, sizeof(info[i].name));
-          sprintf(info[i].name, "wolfSSL_TLS_client_do(%02d)", i);
+        info[i].port = DEFAULT_PORT;
+        info[i].cipher = cipherlist[i];
+        info[i].ctx = client_ctx;
+        info[i].id = i;
 
-          wolfSSL_TLS_client_do(&info[i]);
+        memset(info[i].name, 0, sizeof(info[i].name));
```
Comment (cconlon) [inline IDE/Renesas/e2studio/RA6M4/test/src/test_main.c:409]: Should these calls be changed to our abstracted `XMEMSET` and `XSPRINTF`?

File: IDE/Renesas/e2studio/RA6M4/test/src/wolf_client.c [hunk 2]
```diff
@@ -192,17 +196,17 @@ void wolfSSL_TLS_client_do(void *pvParam)
        /* Set callback CTX */
         #if !defined(TLS_MULTITHREAD_TEST)
         
-        memset(&guser_PKCbInfo, 0, sizeof(User_SCEPKCbInfo));
+        memset(&guser_PKCbInfo, 0, sizeof(FSPSM_ST));
```
Comment (cconlon) [inline IDE/Renesas/e2studio/RA6M4/test/src/wolf_client.c:199]: Let's also change these `memset` calls to `XMEMSET` for better portability.

File: wolfssl/wolfcrypt/port/Renesas/renesas-fspsm-types.h [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+#ifndef __RENESAS_FSP_CRYPT_H__
+#define __RENESAS_FSP_CRYPT_H__
+
+#include <wolfssl/wolfcrypt/settings.h>
+#include <wolfssl/wolfcrypt/logging.h>
+
+
+#if defined(WOLFSSL_RENESAS_SCEPROTECT)
+
+#    include "r_sce.h"
```
Comment (cconlon) [inline wolfssl/wolfcrypt/port/Renesas/renesas-fspsm-types.h:30]: Please fix spacing on this `#include`, thanks

**Review by cconlon — CHANGES_REQUESTED** [review]

File: IDE/Renesas/e2studio/RA6M4/common/user_settings.h [hunk 4]
```diff
@@ -18,6 +18,32 @@
...
  */
+#if defined(WOLFSSL_RENESAS_SCEPROTECT)
+    /* FSP SM stands for Flexible Software Package Security Module
+    *  WOLFSSL_RENESAS_FSPSM enables fundamental code when it uses.
+    *    e.g. Open/Close/Random generator
+    *  WOLFSSL_RENESAS_FSPSPM_TLS enables TLS related code for FSP SM
+    *    e.g. Certificate verification, Master Secret Generation
+    *  WOLFSSL_RENESAS_SCEPROTECT enables specific code for SCE if needed
+    */
+#   define WOLFSSL_RENESAS_FSPSM
```
[the remark this answers is not in the document]
Comment (miyazakh) [reply IDE/Renesas/e2studio/RA6M4/common/user_settings.h]: Thanks, Fixed

File: IDE/Renesas/e2studio/RA6M4/test/src/test_main.c (hunk 1 above)
Comment (miyazakh) [reply IDE/Renesas/e2studio/RA6M4/test/src/test_main.c]: Agree, thanks, Fixed.

File: IDE/Renesas/e2studio/RA6M4/test/src/wolf_client.c (hunk 2 above)
Comment (miyazakh) [reply IDE/Renesas/e2studio/RA6M4/test/src/wolf_client.c]: Thanks, Fixed.

File: wolfssl/wolfcrypt/port/Renesas/renesas-fspsm-types.h (hunk 3 above)
Comment (miyazakh) [reply wolfssl/wolfcrypt/port/Renesas/renesas-fspsm-types.h]: Thanks, fixed.

**Comment by miyazakh** [comment]
retest this please

**Review by cconlon — APPROVED** [review]

**Pull request merged**